Foundation/NSTask.swift

```diff
@@ -37,7 +117,64 @@ public class NSTask : NSObject {
     public var standardError: AnyObject?
     
     // actions
-    public func launch() { NSUnimplemented() }
+    public func launch() {
+    
+        // Dispatch the run loop if it isn't already running
+        
+        pthread_once(&runLoopSetupOnceToken, runLoopSetup)
+        
+        // Ensure that the launch path is set
+        
+        guard let launchPath = self.launchPath else {
+            fatalError()
+        }
+        
+        // Convert the arguments array into a posix_spawn-friendly format
+        
+        var args = [launchPath.lastPathComponent]
+        if let arguments = self.arguments {
+            args.appendContentsOf(arguments)
+        }
+        
+        let argv : UnsafeMutablePointer<UnsafeMutablePointer<Int8>> = args.withUnsafeBufferPointer {
+            let array : UnsafeBufferPointer<String> = $0
+            let buffer = UnsafeMutablePointer<UnsafeMutablePointer<Int8>>.alloc(array.count + 1)
+            buffer.initializeFrom(array.map { $0.withCString(strdup) })
+            buffer[array.count] = nil
+            return buffer
+        }
+        
+        defer {
+            for arg in argv ..< argv + args.count {
+                free(UnsafeMutablePointer<Void>(arg.memory))
+            }
+            
+            argv.dealloc(args.count + 1)
+        }
+        
+        // Lock...
+        
+        processLaunchLock.lock()
+        
+        // ...and load
+        
+        var pid = pid_t()
+        let status = posix_spawn(&pid, launchPath, nil, nil, argv, nil)
```

Comment:
a version 0 run loop source should be created here and added to the CFRunLoop in default mode